components/sup/src/manager/census.rs

```diff
@@ -301,12 +301,25 @@ impl CensusEntry {
         if sg.organization.is_some() {
             self.set_org(sg.organization.unwrap().clone());
         }
-        self.set_ip(String::from(service_rumor.get_ip()));
-        self.set_hostname(String::from(service_rumor.get_hostname()));
-        self.set_port(format!("{}", service_rumor.get_port()));
-        self.set_exposes(service_rumor.get_exposes().iter().map(|p| format!("{}", p)).collect());
-        self.set_package_ident(PackageIdent::from_str(service_rumor.get_package_ident())
+        self.set_ip(String::from(rumor.get_ip()));
+        self.set_hostname(String::from(rumor.get_hostname()));
+        self.set_port(format!("{}", rumor.get_port()));
+        self.set_exposes(rumor.get_exposes().iter().map(|p| format!("{}", p)).collect());
+        // JW TODO: we can't crash the supervisor because we've received bad data. This could have
+        // gotten to us from a supervisor speaking a slightly different protocol or from a
+        // malicious service speaking our gossip protocol. We should consider ignoring this update
+        // altogether perhaps? We can't just unload the service because that would also give an
+        // attacker what they want.
+        self.set_package_ident(PackageIdent::from_str(rumor.get_package_ident())
```

Comment:
So we should just not update our census entry, yeah? That's what I was thinking but I wanted to make sure